### PR TITLE
Fix for #4047

### LINKED ISSUE #4047: WARNING: Invalid bid sent to bidder quantcast:
quantcast seems to be throwing this warning, where as all other bids i have don't, when using multi format boxes

WARNING: Invalid bid sent to bidder quantcast: {"bidder":"quantcast","params":{"publisherId":"XXXXXXX"},"crumbs":{"pubcid":"fdccbcf6-8042-459d-b72e-6f1f8ea123a4"},"nativeParams":{"image":{"required":true,"sizes":[150,50]},"title":{"required":true,"len":80},"sponsoredBy":{"required":true},"clickUrl":{"required":true},"privacyLink":{"required":false},"body":{"required":true},"icon":{"required":true,"sizes":[50,50]}},"renderer":{"options":{"adText":"Advertisement"}},"mediaTypes":{"banner":{"sizes":[[300,250],[728,90],[250,250],[468,60],[320,50]]},"native":{"image":{"required":true,"sizes":[150,50]},"title":{"required":true,"len":80},"sponsoredBy":{"required":true},"clickUrl":{"required":true},"privacyLink":{"required":false},"body":{"required":true},"icon":{"required":true,"sizes":[50,50]}},"video":{"context":"outstream","playerSize":[[550,310]],"mimes":["video/x-flv","video/mp4","application/x-shockwave-flash","application/javascript","video/webm"],"playbackmethod":[2],"minduration":0,"maxduration":30,"protocols":[2,3],"linearity":1,"api":[2]}},"adUnitCode":"inreedvid5Slot","transactionId":"c3359a31-4e4c-44ae-8a40-1bd6ab70dbdf","sizes":[[300,250],[728,90],[250,250],[468,60],[320,50]],"bidId":"43d09b8bb90ffce","bidderRequestId":"41e0128c5ac0e3f","auctionId":"bb8e5c29-2c4c-46a8-9bfd-4fa748caf184","src":"client","bidRequestsCount":1}

### PR DESCRIPTION
**Pull request by dpapworth-qc**
## Type of change

- [x] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Resolves #4047.
Change isBidRequestValid to reject bid requests missing required parameters, rather than unsupported media types.
Filter unsupported media types when building requests.
Add test for multi-format ad units to verify expected behaviour.

### CODE REVIEW

**Comment by dpapworth-qc** [comment]
Build failure appears to be due to an error when running RealVu Analytics Adapter tests.

File: modules/quantcastBidAdapter.js [hunk 1]
```diff
@@ -112,12 +102,21 @@ export const spec = {
...
-      const videoContext = utils.deepAccess(bid, 'mediaTypes.video.context');
-      if (videoContext === 'instream') {
-        imp = makeVideoImp(bid);
+      if (bid.mediaTypes) {
+        if (bid.mediaTypes.video && bid.mediaTypes.video.context === 'instream') {
+          imp = makeVideoImp(bid);
+        } else if (bid.mediaTypes.banner) {
+          imp = makeBannerImp(bid);
+        } else {
+          // Unsupported mediaType
```
Comment (jsnellbaker) [inline modules/quantcastBidAdapter.js]: Do you want to throw a warning/info message message here to note that an unsupported bid type was ignored by the adapter (if only for future diagnostic information)?

**Comment by jsnellbaker** [comment]
@dpapworth-qc As an update, I reran the circleci tests.  They should be good.  

The current failure (where the job seemed to stall) was fixed in a recent update to master.  If you resync your branch and get the update; it should resolve the issue.

File: modules/quantcastBidAdapter.js (hunk 1 above)
Comment (dpapworth-qc) [reply modules/quantcastBidAdapter.js]: :+1: done

**Comment by dpapworth-qc** [comment]
@jsnellbaker i've added a log message for unsupported media types, and i've merged the current master branch.

**Pull request merged**